tools/FeynOpt.hs

```diff
@@ -213,38 +263,51 @@ printHelp = mapM_ putStrLn lines
           ]
 
 
-parseArgs :: [Pass] -> Bool -> Bool -> [String] -> IO ()
-parseArgs passes verify pureCircuit []     = printHelp
-parseArgs passes verify pureCircuit (x:xs) = case x of
+data Options = Options {passes :: [Pass], verify :: Bool, pureCircuit :: Bool, useQASM3 :: Bool}
```

Comment:
Not sure how idiomatic this is, but I like the recursive parseArgs structure just fine, and switching to a structured data from loose booleans pretty much alleviates the problems with adding/removing state in the parser.